src/zl-util/ZLInterpolate.cpp

```diff
@@ -162,6 +164,154 @@ float ZLInterpolate::Curve ( u32 mode, float t ) {
 		
 			return Curve ( kSoftSmooth, _pow ( t ));
 
+		//................................................................
+		case kSineEaseIn:
+			
+			return sinf ( t * M_PI_2 );
+		
+		//................................................................
+		case kSineEaseOut:
+			
+			return 1.0f - cosf ( t * M_PI_2 );
+
+		//................................................................
+		case kSineSmooth:
+			
+			return 0.5f - 0.5f * cosf ( t * M_PI );
+
+		//................................................................
+		case kCircEaseIn:
+			
+			t = t - 1.0f;
+			return sqrtf ( 1.0f - t * t );
+
+		//................................................................
+		case kCircEaseOut:
+			
+			return 1.0f - sqrtf ( 1.0f - t * t );
+
+		//................................................................
+		case kCircSmooth:
+			
+			if ( t < 0.5f ) {
+				t = t * 2.0f;
+				return 0.5f - 0.5f * sqrtf ( 1.0f - t * t );
+			}
+			t = t * 2.0f - 2.0f;
+			return 0.5f + 0.5f * sqrtf ( 1.0f - t * t );
+			
+		//................................................................
+		case kBounceIn:
+			
+			if ( t < (1.0f / 2.75f) ) {
+				return 7.5625f * t * t;
+			}
+			else if ( t < (2.0f / 2.75f) ) {
+				t = t - 1.5f / 2.75f;
+				return 7.5625f * t * t + 0.75f;
+			}
+			else if ( t < (2.5f / 2.75f) ) {
+				t = t - 2.25f / 2.75f;
+				return 7.5625f * t * t + 0.9375f;
+			}
+			else {
+				t = t - 2.625 / 2.75f;
+				return 7.5625f * t * t + 0.984375f;
+			}
+			return 0.0f;
+
+		//................................................................
+		case kBounceOut:
+			
+			return 1.0f - Curve ( kBounceIn, 1.0f - t );
+
+		//................................................................
+		case kBounceSmooth:
+			
+			if ( t < 0.5f ) {
+				return 0.5f * Curve ( kBounceOut, t * 2.0f );
+			}
+			else {
+				return 0.5f + 0.5f * Curve ( kBounceIn, 2.0f * t - 1.f );
+			}
+			
```

Comment:
`BOUNCE_SMOOTH` stops and re-accelarates in the middle:
![bounce_smooth](https://cloud.githubusercontent.com/assets/5963162/5605739/91e25d1a-940d-11e4-8c98-5822357a8747.png)
Is this on purpose? I believe it would look better if the middle part was one smooth motion, as in `ELASTIC_SMOOTH`.
![elastic_smooth](https://cloud.githubusercontent.com/assets/5963162/5605743/ae2d88dc-940d-11e4-8aa9-9f996e37a10a.png)